dev/src/watch.ts

```diff
@@ -426,50 +427,55 @@ abstract class Watch {
      * Initializes a new stream to the backend with backoff.
      */
     const initStream = () => {
-      this._backoff.backoffAndWait().then(async () => {
-        if (!isActive) {
-          logger(
-            'Watch.onSnapshot',
-            this._requestTag,
-            'Not initializing inactive stream'
-          );
-          return;
-        }
+      this._backoff
+        .backoffAndWait()
+        .catch(err => {
+          closeStream(err);
```

Comment:
After you deal with a Promise in `.catch()` it is no longer considered rejected. While an error in the backoff timer is handled by closing the stream, the code as written then immediately tries to re-establish it. This doesn't happen here because `closeStream()` sets `isActive` to false, but I would prefer if we did not rely on this.

Reply:
By this, you mean to pass along the Promise rejection right?

Reply:
If you just revert your change and call `.then` before `.catch`, everything will work again. Otherwise you have to do:

```
.catch(e => {
   ...
   return Promise.reject(e);
}
```